contracts/ERC721A.sol

```diff
@@ -352,13 +413,12 @@ contract ERC721A is Context, ERC165, IERC721, IERC721Metadata, IERC721Enumerable
                 if (safe && !_checkOnERC721Received(address(0), to, updatedIndex, _data)) {
                     revert TransferToNonERC721ReceiverImplementer();
                 }
-
                 updatedIndex++;
             }
 
-            _currentIndex = updatedIndex;
+            if (updatedIndex > type(uint128).max) revert SafecastOverflow();
```

Comment:
I think the same, also I think all the changes in the `_mint` function should be in another PR if they are irrelevant to burn functionality.

Reply:
I think we can remove the overflow check for now. 

The overflow check is placed there in preparation for customisable starting index (in case someone sets the starting index too high). 

Instead of the check, we can leave a comment warning users to check that their customised startIndex is within safe bounds. Most users will either use 0 or 1 as the startIndex, so overflow is still very unrealistic.

I’m ok with either uint128 or uint256 for this PR.